### PR TITLE
QE: Add sync of custom channels for Rocky 8

### PR DESCRIPTION
**Pull request by nodeg**
## What does this PR change?

This will add a sync of Rocky 8 via the command line and fixes some of the issues currently present in the CI.

PR test run: https://ci.suse.de/view/Manager/view/Uyuni-PRs/job/uyuni-prs-ci-tests/1839

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage

- Cucumber tests were edited

- [x] **DONE**

## Links

fixes https://github.com/SUSE/spacewalk/issues/19322
no ports needed.

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by elariekerboull — APPROVED** [review]
Lgtm

File: testsuite/features/reposync/srv_add_rocky8_repositories.feature [hunk 1]
```diff
@@ -50,6 +50,11 @@ Feature: Add the Rocky 8 distribution custom repositories
     And I click on "Save Repositories"
     Then I should see a "repository information was successfully updated" text
 
+  Scenario: Synchronize the repositories in the custom channel for Rocky 8 DVD
+    When I call spacewalk-repo-sync for channel "rocky-8-iso"
+    Then I wait until all spacewalk-repo-sync finished
```
Comment (Bischoff) [inline testsuite/features/reposync/srv_add_rocky8_repositories.feature:55]: And

File: testsuite/features/reposync/srv_add_rocky8_repositories.feature [hunk 2]
```diff
@@ -50,6 +50,11 @@ Feature: Add the Rocky 8 distribution custom repositories
...
+    And the reposync logs should not report errors
```
Comment (Bischoff) [inline testsuite/features/reposync/srv_add_rocky8_repositories.feature:56]: Then

**Review by Bischoff — CHANGES_REQUESTED** [review]
nitpicks

File: testsuite/features/reposync/srv_add_rocky8_repositories.feature (hunk 1 above)
Comment (nodeg) [reply testsuite/features/reposync/srv_add_rocky8_repositories.feature]: Fixed.

File: testsuite/features/reposync/srv_add_rocky8_repositories.feature (hunk 2 above)
Comment (nodeg) [reply testsuite/features/reposync/srv_add_rocky8_repositories.feature]: Fixed.

**Review by Bischoff — APPROVED** [review]

**Review by srbarrios — APPROVED** [review]

File: testsuite/features/reposync/srv_add_rocky8_repositories.feature [hunk 3]
```diff
@@ -50,6 +50,11 @@ Feature: Add the Rocky 8 distribution custom repositories
     And I click on "Save Repositories"
     Then I should see a "repository information was successfully updated" text
 
+  Scenario: Synchronize the repositories in the custom channel for Rocky 8 DVD
+    When I call spacewalk-repo-sync for channel "rocky-8-iso"
```
Comment (srbarrios) [inline testsuite/features/reposync/srv_add_rocky8_repositories.feature:54]: The method above, where we can pass a custom url, it's about the custom repo?
Maybe we can generate both repos plus sync the channel all in one command line execution? 🤔

Comment (srbarrios) [reply testsuite/features/reposync/srv_add_rocky8_repositories.feature:54]: In any case, LGTM, you can iterate over it if you want later.

Comment (nodeg) [reply testsuite/features/reposync/srv_add_rocky8_repositories.feature:54]: There was a method with a custom URL. 

```ruby
When(/^I call spacewalk\-repo\-sync for channel "(.*?)" with a custom url "(.*?)"$/) do |arg1, arg2|
  @command_output, _code = $server.run("spacewalk-repo-sync -c #{arg1} -u #{arg2}", check_errors: false)
end
```

Comment (srbarrios) [reply testsuite/features/reposync/srv_add_rocky8_repositories.feature:54]: Yup, the method above the one I marked :)
I'm suggesting to refactor it to support multiple custom repos.

File: testsuite/features/step_definitions/command_steps.rb [hunk 4]
```diff
@@ -741,6 +741,10 @@
   @command_output, _code = $server.run("spacewalk-repo-sync -c #{arg1} -u #{arg2}", check_errors: false)
 end
 
+When(/^I call spacewalk\-repo\-sync for channel "(.*?)"$/) do |channel|
```
Comment (elariekerboull) [inline testsuite/features/step_definitions/command_steps.rb:744]: This step description creates an ambiguity with the above step (`When(/^I call spacewalk\-repo\-sync for channel "(.*?)" with a custom url "(.*?)"$/) do |arg1, arg2|`), a simple change could solve this (e.g. `to sync the channel`)

**Comment by nodeg** [comment]
I still have to take into account that the client tools for RES8 have to be synced differently on Uyuni than on SUMA. Working on it.

File: testsuite/features/step_definitions/common_steps.rb [hunk 5]
```diff
@@ -15,7 +15,12 @@
 end
 
 When(/^I mount as "([^"]+)" the ISO from "([^"]+)" in the server$/) do |name, url|
-  iso_path = "/tmp/#{name}.iso"
+  # When using a mirror it is automatically mounted at /mirror
+  iso_path = if $mirror
+               url.sub(/^http:.*\/pub/, '/mirror/pub')
+             else
+               "/tmp/#{name}.iso"
+             end
```
Comment (srbarrios) [inline testsuite/features/step_definitions/common_steps.rb:23]: Are you sure we can mount using as path a mounted file?

File: testsuite/features/step_definitions/common_steps.rb [hunk 6]
```diff
@@ -15,7 +15,12 @@
...
   mount_point = "/srv/www/htdocs/#{name}"
   $server.run("wget --no-check-certificate -O #{iso_path} #{url}", timeout: 500)
```
Comment (srbarrios) [inline testsuite/features/step_definitions/common_steps.rb:25]: We don't need this line, if we use the mirror as refactored above.

File: testsuite/features/step_definitions/common_steps.rb (hunk 5 above)
Comment (nodeg) [reply testsuite/features/step_definitions/common_steps.rb:23]: According to @Bischoff mounting something inside a mount is not problem. Same goes for the root (`/`) file system Eric explained to me.

File: testsuite/features/step_definitions/common_steps.rb (hunk 6 above)
Comment (nodeg) [reply testsuite/features/step_definitions/common_steps.rb]: Fixed.

**Comment by nodeg** [comment]
ATM the scenario to create bootstrap repositories is failing due to missing packages:

```bash
19:28:48  Scenario: Create the bootstrap repository for the Rocky 8 minion           # features/reposync/srv_add_rocky8_repositories.feature:112
19:28:48  This scenario ran at: 2022-10-21 19:28:31 +0200
19:28:48  When I create the bootstrap repository for "rhlike_minion" on the server # features/step_definitions/command_steps.rb:944
19:28:48  Creating the boostrap repository on the server:
19:28:48    mgr-create-bootstrap-repo --create SLE-ES8-x86_64 --with-custom-channels --flush
19:28:48        FAIL: mgr-create-bootstrap-repo --create SLE-ES8-x86_64 --with-custom-channels --flush returned status code = 1.
19:28:48        Output:
19:28:48        Directory walk started
19:28:48        Directory walk done - 15 packages
19:28:48        Temporary output repo path: /srv/www/htdocs/pub/repositories/res/8/bootstrap.tmp/.repodata/
19:28:48        Preparing sqlite DBs
19:28:48        Pool started (with 5 workers)
19:28:48        Pool finished
19:28:48        ERROR: package 'salt' not found
19:28:48        ERROR: package 'salt-minion' not found
19:28:48        ERROR: package 'python3-salt' not found
19:28:48        ERROR: package 'python3-msgpack' not found
19:28:48        ERROR: package 'python3-tornado' not found
19:28:48        ERROR: package 'python3-zmq' not found
19:28:48        ERROR: package 'python3-m2crypto' not found
19:28:48        ERROR: package 'openpgm' not found
19:28:48        ERROR: package 'zeromq' not found
19:28:48        ERROR: package 'python3-immutables' not found
19:28:48        ERROR: package 'python3-contextvars' not found
19:28:48        ERROR: package 'venv-salt-minion' not found
19:28:48        
19:28:48        Suggestions:
19:28:48        mgr-create-bootstrap-repo uses the locally synchronized versions of files
19:28:48        from the Tools repository, and uses the locally synchronized pool channel
19:28:48        for dependency resolution.
19:28:48        Both should be fully synced before running the mgr-create-bootstrap-repo script.
19:28:48        
19:28:48        
19:28:48        Creating bootstrap repo for SLE-ES8-x86_64
19:28:48        copy 'redhat-rpm-config-129-1.el8.noarch'
19:28:48        copy 'python3-babel-2.5.1-7.el8.noarch'
19:28:48        copy 'python3-jinja2-2.10.1-3.el8.noarch'
19:28:48        copy 'python3-markupsafe-0.23-19.el8.x86_64'
19:28:48        copy 'python3-psutil-5.4.3-11.el8.x86_64'
19:28:48        copy 'python3-pyyaml-3.12-12.el8.x86_64'
19:28:48        copy 'python3-requests-2.20.0-2.1.el8_1.noarch'
19:28:48        copy 'python3-urllib3-1.24.2-5.el8.noarch'
19:28:48        copy 'python3-idna-2.5-5.el8.noarch'
19:28:48        copy 'python3-chardet-3.0.4-7.el8.noarch'
19:28:48        copy 'python3-pysocks-1.6.8-3.el8.noarch'
19:28:48        copy 'python3-pytz-2017.2-9.el8.noarch'
19:28:48        copy 'python3-setuptools-39.2.0-6.el8.noarch'
19:28:48        copy 'python3-distro-1.4.0-2.module+el8.3.0+120+426d8baf.noarch'
19:28:48        copy 'dmidecode-3.3-4.el8:1.x86_64'
19:28:48         (RuntimeError)
19:28:48        ./features/support/lavanda.rb:93:in `run'
19:28:48        ./features/step_definitions/command_steps.rb:955:in `/^I create the bootstrap repository for "([^"]*)" on the server$/'
19:28:48        features/reposync/srv_add_rocky8_repositories.feature:113:in `I create the bootstrap repository for "rhlike_minion" on the server'
```

This could happen because we kill the RES/EL8 reposync:

```bash
19:09:16  Scenario: Kill running reposyncs or wait for them to finish                          # features/reposync/srv_wait_for_reposync.feature:16
19:09:16  This scenario ran at: 2022-10-21 18:53:17 +0200
19:09:16  When I kill all running spacewalk-repo-sync, excepted the ones needed to bootstrap # features/step_definitions/command_steps.rb:337
19:09:16  Reposync of channel sles12-sp4-pool-x86_64 killed
19:09:16  Reposync of channel sles12-sp4-pool-x86_64 killed
19:09:16  Reposync of channel sles12-sp4-pool-x86_64 killed
19:09:16  Reposync of channel sles12-sp4-pool-x86_64 killed
19:09:16  Reposync of channel sles12-sp4-pool-x86_64 killed
19:09:16  Reposync of channel sles12-sp4-pool-x86_64 killed
19:09:16  Reposync of channel sles12-sp4-updates-x86_64 killed
19:09:16  Reposync of channel sles12-sp5-pool-x86_64 killed
19:09:16  Reposync of channel sles12-sp5-updates-x86_64 killed
19:09:16  Reposync of channel sles12-sp5-installer-updates-x86_64 killed
19:09:16  Reposync of channel res-as-8-updates-x86_64 killed
19:09:16  Reposync of channel res-cb-8-updates-x86_64 killed
19:09:16  Reposync of channel sle-product-sles_sap15-pool-x86_64 killed
19:09:16  Reposync of channel sle-product-sles_sap15-updates-x86_64 killed
19:09:16  Reposync of channel sle-product-sles15-sp4-pool-x86_64 left running
19:09:16  Reposync of channel sle15-sp4-installer-updates-x86_64 killed
19:09:16  Reposync of channel sle-product-sles_sap15-updates-x86_64 killed
19:09:16  Reposync of channel sle-product-sles_sap15-pool-x86_64 killed
19:09:16  Reposync of channel sle-module-basesystem15-sp4-updates-x86_64 left running
19:09:16  Reposync of channel sle-module-basesystem15-sp4-updates-x86_64 left running
19:09:16  Reposync of channel sle-module-basesystem15-sp4-updates-x86_64 left running
19:09:16  Reposync of channel sle-module-basesystem15-sp4-updates-x86_64 left running
19:09:16  Reposync of channel sle-module-basesystem15-sp4-pool-x86_64 left running
19:09:16  Reposync of channel sle-module-basesystem15-sp4-pool-x86_64 left running
19:09:16  Reposync of channel sle-module-basesystem15-sp4-pool-x86_64 left running
19:09:16  Reposync of channel sle-module-basesystem15-sp4-pool-x86_64 left running
19:09:16  Reposync of channel rhel8-pool-x86_64 killed
19:09:16  Reposync of channel sle-module-legacy12-pool-x86_64-sp5 killed
19:09:16  Reposync of channel sle-module-legacy12-updates-x86_64-sp5 killed
19:09:16  Reposync of channel res-8-updates-x86_64 killed
19:09:16  Reposync of channel res-cb-8-updates-x86_64 killed
19:09:16  Reposync of channel res-as-8-updates-x86_64 killed
19:09:16  Reposync of channel el8-uyuni-client-x86_64 killed
19:09:16  Reposync of channel sle-module-server-applications15-sp4-pool-x86_64 left running
19:09:16  Reposync of channel sle-module-server-applications15-sp4-updates-x86_64 left running
19:09:16  Reposync of channel sle-module-containers12-pool-x86_64-sp4 killed
19:09:16  Reposync of channel sle-module-containers12-updates-x86_64-sp4 killed
19:09:16  Reposync of channel sle-product-sled15-sp3-updates-x86_64 killed
19:09:16  Reposync of channel sle-product-sled15-sp3-pool-x86_64 killed
19:09:16  Reposync of channel sle-module-basesystem15-sp3-pool-x86_64-sled killed
19:10:36  Reposync of channel sle-module-basesystem15-sp3-updates-x86_64-sled killed
19:10:36  Reposync of channel sle-module-desktop-applications15-sp3-pool-x86_64-sled killed
19:10:36  Reposync of channel sle-module-desktop-applications15-sp3-updates-x86_64-sled killed
19:10:36  Reposync of channel sle-module-python2-15-sp3-pool-x86_64-sled killed
19:10:36  Reposync of channel sle-module-python2-15-sp3-updates-x86_64-sled killed
19:10:36  Reposync of channel sle-product-we15-sp3-updates-x86_64-sled killed
19:10:36  Reposync of channel sle-product-we15-sp3-pool-x86_64-sled killed
19:10:36  Reposync of channel sle-15-sp3-desktop-nvidia-driver killed
19:10:36  Reposync of channel sle-product-sled15-sp3-pool-x86_64 killed
19:10:36  Reposync of channel sle-product-sled15-sp3-updates-x86_64 killed
19:10:36  This scenario took: 950 seconds
```

**Comment by nodeg** [comment]
I sadly still see the following during the PR test although it should not be killed.

```bash
15:41:34  Reposync of channel el8-uyuni-client-x86_64 killed
```

**Comment by nodeg** [comment]
Rebase done.

**Comment by nodeg** [comment]
RuboCop complains:

```bash
Style/MultipleComparison: Avoid comparing a variable with multiple items in a conditional, use Array#include? instead
```

File: testsuite/features/secondary/min_rhlike_remote_command.feature [hunk 7]
```diff
@@ -19,5 +19,5 @@ Feature: Remote command on the Red Hat-like Salt minion
     Then I should see "rhlike_minion" hostname
     When I wait for "15" seconds
     And I expand the results for "rhlike_minion"
-    Then I should see a "rhel fedora" text
-    And I should see a "REDHAT_SUPPORT_PRODUCT" text
+    Then I should see a "rhel centos fedora" text
```
Comment (srbarrios) [inline testsuite/features/secondary/min_rhlike_remote_command.feature:22]: Centos?

**Review by srbarrios — APPROVED** [review]
Quick look, but it looks good.
Just one doubt in one line, see my pr comment.

File: testsuite/features/secondary/min_rhlike_remote_command.feature (hunk 7 above)
Comment (nodeg) [reply testsuite/features/secondary/min_rhlike_remote_command.feature:22]: That proposed change is correct. See the output of `/etc/os-release`:

```bash

Last login: Mon Oct 24 12:57:49 2022
[root@rocky-min-rocky8 ~]# cat /etc/os-release 
NAME="Rocky Linux"
VERSION="8.6 (Green Obsidian)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="8.6"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Rocky Linux 8.6 (Green Obsidian)"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:rocky:rocky:8:GA"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
ROCKY_SUPPORT_PRODUCT="Rocky Linux"
ROCKY_SUPPORT_PRODUCT_VERSION="8"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
```

**Review by elariekerboull — APPROVED** [review]

**Review by ktsamis — APPROVED** [review]

**Comment by nodeg** [comment]
Thanks for the review. I am rebasing the PR and then merge it.

**Pull request merged**